pkg/controller/logstorage/logstorage_controller.go

```diff
@@ -666,58 +797,133 @@ func (r *ReconcileLogStorage) getElasticsearchCertificateSecrets(ctx context.Con
+	esKeyCert, err := utils.GetSecret(ctx, r.client, render.TigeraElasticsearchInternalCertSecret, render.ElasticsearchNamespace)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Ensure that cert is valid.
+	esKeyCert, err = utils.EnsureCertificateSecret(render.TigeraElasticsearchInternalCertSecret, esKeyCert, corev1.TLSPrivateKeyKey, corev1.TLSCertKey, rmeta.DefaultCertificateDuration, svcDNSNames...)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	// Override the Operator namespace set by utils.EnsureCertificateSecret.
+	esKeyCert.Namespace = render.ElasticsearchNamespace
+
+	// Three different certificate issuers are possible:
+	// - The operator self-signed certificate
+	// - A user's BYO keypair for Elastic (uncommon)
+	// - The issuer that is provided through the certificate management feature.
+	keyCertIssuer, err := utils.GetCertificateIssuer(esKeyCert.Data[corev1.TLSCertKey])
+	if err != nil {
+		return nil, nil, err
+	}
+	customerProvidedCert := !utils.IsOperatorIssued(keyCertIssuer)
+
+	// If Certificate management is enabled, we only want to trust the CA cert andlet the init container handle private key generation.
+	if instl.CertificateManagement != nil {
+		cmCa := instl.CertificateManagement.CACert
+		cmIssuer, err := utils.GetCertificateIssuer(cmCa)
+		if err != nil {
+			return nil, nil, err
+		}
+
+		// If the issuer of the current secret is not the same as the certificate management issuer and also is not
+		// issued by the tigera-operator, it means that it is added to this cluster by the customer. This is not supported
+		// in combination with certificate management.
+		if customerProvidedCert && cmIssuer != keyCertIssuer {
+			return nil, nil, fmt.Errorf("certificate management does not support custom Elasticsearch secrets, please delete secret %s/%s or disable certificate management", esKeyCert.Namespace, esKeyCert.Name)
+		}
+
+		esKeyCert.Data[corev1.TLSCertKey] = instl.CertificateManagement.CACert
+		certSecret = render.CreateCertificateSecret(instl.CertificateManagement.CACert, relasticsearch.InternalCertSecret, render.ElasticsearchNamespace)
 	} else {
-		esKeyCert = rsecret.CopyToNamespace(render.ElasticsearchNamespace, oprKeyCert)[0]
-		// Get the pub secret - might be nil
-		pubSecret, err := utils.GetSecret(ctx, r.client, relasticsearch.PublicCertSecret, render.ElasticsearchNamespace)
+		// Get the internal public cert secret - might be nil
+		internalSecret, err := utils.GetSecret(ctx, r.client, relasticsearch.InternalCertSecret, render.ElasticsearchNamespace)
 		if err != nil {
-			return nil, nil, nil, err
+			return nil, nil, err
 		}
 
-		if pubSecret != nil {
+		if internalSecret != nil {
 			// If the provided certificate secret (secret) is managed by the operator we need to check if the secret that
 			// Elasticsearch creates from that given secret (pubSecret) has the expected DNS name. If it doesn't, delete the
 			// public secret so it can get recreated.
 			if !customerProvidedCert {
```

Comment:
Could you add a follow up task to evaluate if this route is possible, I don't think it is since you've taken away BYO. Think about it later though, not now, let's try to get this through

Reply:
Sure will do

Reply:
https://tigera.atlassian.net/browse/SAAS-1586